src/console.cpp

```diff
@@ -51,8 +51,8 @@ static void IConsoleWriteToLogFile(const std::string &string)
 {
 	if (_iconsole_output_file != nullptr) {
 		/* if there is an console output file ... also print it there */
-		const char *header = GetLogPrefix();
-		if ((strlen(header) != 0 && fwrite(header, strlen(header), 1, _iconsole_output_file) != 1) ||
+		std::string header = GetLogPrefix();
+		if ((!header.empty() && fwrite(header.c_str(), header.size(), 1, _iconsole_output_file) != 1) ||
 				fwrite(string.c_str(), string.size(), 1, _iconsole_output_file) != 1 ||
 				fwrite("\n", 1, 1, _iconsole_output_file) != 1) {
```

Comment:
```suggestion
		try {
			fmt::print(_iconsole_output_file, "{}{}\n", GetLogPrefix(), string);
		} catch (const std::system_error &e) {
```
What about this solution to writing the output? I checked `format-inl.h`, and that seems to be throwing `std::system_error` when `fwrite` does not write everything.